pkg/provider/cloud/kubevirt/storage.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubevirt
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func EnsureInfraCSIAccess(ctx context.Context, kubeconfig string, clusterName string) ([]byte, error) {
+	// soon we are going to change the default namespace to cluster namespace
+	namespace := "default"
+
+	cli, err := NewClientset(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	sa := &corev1.ServiceAccount{
```

Comment:
Do we know already what kinda resources the client would access? I mean giving access to all resources might impose a security issues.

Reply:
Good idea, was not aware of those.

Reply:
I am not giving an acess to all resources here.